PWGHF/Utils/utilsEvSelHf.h

```diff
@@ -16,82 +16,100 @@
 
 /// @brief Function to put labels on collision monitoring histogram
 /// \param hCollisions is the histogram
 template <typename Histo>
 void setLabelHistoEvSel(Histo& hCollisions)
 {
-  hCollisions->GetXaxis()->SetBinLabel(ValuesEvSel::All + 1, "All collisions");
-  hCollisions->GetXaxis()->SetBinLabel(ValuesEvSel::Cent + 1, "Centrality ok");
-  hCollisions->GetXaxis()->SetBinLabel(ValuesEvSel::CentSel8 + 1, "Centrality + sel8 ok");
-  hCollisions->GetXaxis()->SetBinLabel(ValuesEvSel::CentSel8PosZ + 1, "Centrality + sel8 + posZ ok");
-  hCollisions->GetXaxis()->SetBinLabel(ValuesEvSel::CentSel8PosZTFBorder + 1, "Centrality + sel8 + posZ + TF border ok");
+  hCollisions->SetTitle("HF event counter;;accepted collisions");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::None + 1, "All collisions");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::Centrality + 1, "Centrality");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::Trigger + 1, "Trigger");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::TimeFrameBorderCut + 1, "TF border");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::NContrib + 1, "# of PV contributors");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::Chi2 + 1, "PV #it{#chi}^{2}");
+  hCollisions->GetXaxis()->SetBinLabel(EventRejection::PositionZ + 1, "PV #it{z}");
 }
 
 /// \brief Function to apply event selections in HF analyses
 /// \param collision collision that has to satisfy the selection criteria
+/// \param centrality collision centrality to be initialised in this function
+/// \param centralityMin minimum centrality accepted
+/// \param centralityMin maximum centrality accepted
 /// \param useSel8Trigger switch to activate the sel8() event selection
+/// \param triggerClass trigger class different from sel8 (e.g. kINT7 for Run2) used only if useSel8Trigger is false
+/// \param zPvPosMin minimum primary-vertex z
 /// \param zPvPosMax maximum primary-vertex z
+/// \param nPvContributorsMin minimum number of PV contributors
+/// \param chi2PvMax maximum PV chi2
 /// \param useTimeFrameBorderCut switch to activate the time frame border cut
 /// \return a bitmask with the event selections not satisfied by the analysed collision
-template <o2::aod::hf_collision_centrality::CentralityEstimator centEstimator, typename Coll>
-uint16_t getHfCollisionRejectionMask(const Coll& collision, float centralityMin, float centralityMax, bool useSel8Trigger, float maxPvPosZ, bool useTimeFrameBorderCut)
+template <bool applyEvSel, o2::aod::hf_collision_centrality::CentralityEstimator centEstimator, typename Coll>
+uint16_t getHfCollisionRejectionMask(const Coll& collision, float& centrality, float centralityMin, float centralityMax, bool useSel8Trigger, int triggerClass, bool useTimeFrameBorderCut, float zPvPosMin, float zPvPosMax, int nPvContributorsMin, float chi2PvMax)
```

Comment:
Please add documentation of `applyEvSel` and `centEstimator` and sort the `\param` lines to match the order of appearance in the function signature.

Reply:
Done